tests/phpunit/tests/kses.php

```diff
@@ -1362,8 +1362,8 @@ public function data_safecss_filter_attr() {
 	 * @ticket 33121
 	 */
 	public function test_wp_kses_attr_data_attribute_is_allowed() {
-		$test     = '<div data-foo="foo" data-bar="bar" datainvalid="gone" data--invalid="gone"  data-also-invalid-="gone" data-two-hyphens="remains">Pens and pencils</div>';
-		$expected = '<div data-foo="foo" data-bar="bar" data-two-hyphens="remains">Pens and pencils</div>';
+		$test     = '<div data-foo="foo" data-bar="bar" datainvalid="gone" data--double-dash="retained"  data-trailing-dash-="allowable" data-two-hyphens="remains">Pens and pencils</div>';
+		$expected = '<div data-foo="foo" data-bar="bar" data--double-dash="retained" data-trailing-dash-="allowable" data-two-hyphens="remains">Pens and pencils</div>';
```

Comment:
I'd like to see cases covering `data--leading` `data-trailing-` and `data-middle--double` cases. The `--` together in the middle is missing.

Reply:
With this change it would be good to convert this to use a data provider and add a bunch more valid and invalid use cases.

It probably should have been a data provider all along but please do not look in to the history of this feature to figure out who didn't do that in the first place. ;)